### PR TITLE
[local-authentication][android] Fix promise won't be resolved if the user used a PIN

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Fixed promise won't be resolved if the user used a PIN as an auth method. 

# How

- Passes data from `onActivityResult` to the underlying auth fragment to trigger the `AuthenticationCallback`.

# Test Plan

- bare-expo ✅

### CODE REVIEW

**Review by byCedric — APPROVED** [review]
Looks good to me, maybe nice to have a NCL/test-suite entry for this? So we can catch this during QA next time? 😄

**Review by mczernek — APPROVED** [review]

**Comment by lukmccall** [comment]
We have a flow in NCL that cover this case already :D 
But this bug only occurs on some Android devices.

File: packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.java [hunk 1]
```diff
@@ -240,6 +244,25 @@ public void run() {
     });
   }
 
+  @Override
+  public void onActivityResult(Activity activity, int requestCode, int resultCode, Intent data) {
```
Comment (shubhamdeol) [inline packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.java:248]: have you tested this on android devices on android 7, 8 and 9.
Promise is not being resolved on such devices when calling .
``` LocalAuthentication.authenticateAsync```
I had to rollback to old version(9.2.0) when things worked fine

Comment (torufuruya) [reply packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.java:248]: Same here

**Pull request merged**